### PR TITLE
Add Linux support to Package Info Plus

### PR DESCRIPTION
**Pull request by jpnurmi**
Requires https://github.com/flutter/flutter/pull/69920

### CODE REVIEW

**Comment by jpnurmi** [comment]
Requires https://github.com/flutter/flutter/pull/69920

**Comment by mhadaily** [comment]
Seems like the build is failing on this @jpnurmi

**Comment by jpnurmi** [comment]
Yeah it's still work in progress so I didn't care about the analysis failure. Let's see if they accept the patch to Flutter, first. The proposal is using the same version.json approach as on web.

P.S. We should publish all the unpublish packages because there are a few dependencies marked as ^0.1.0 here and there, but melos cannot handle it if the package doesn't exist at all. Pub get fails in the first place so melos has no chance to fix up pubspec.lock.

**Comment by jpnurmi** [comment]
Or perhaps the problem is here since the CI build passes... 😁

File: packages/package_info_plus_linux/lib/src/package_info.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/// The Linux implementation of [PackageInfoPlatform].
+class PackageInfoLinux extends PackageInfoPlatform {
+  /// Returns a map with the following keys:
+  /// appName, packageName, version, buildNumber
+  @override
+  Future<PackageInfoData> getAll() async {
+    final exePath = File('/proc/self/exe').resolveSymbolicLinksSync();
+    final appPath = path.dirname(exePath);
+    final assetPath = path.join(appPath, 'data', 'flutter_assets');
+    final versionPath = path.join(assetPath, 'version.json');
```
Comment (jpnurmi) [inline packages/package_info_plus_linux/lib/src/package_info.dart:17]: handle non-existent version.json

File: packages/package_info_plus_linux/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: package_info_plus_linux
+description: Linux implementation of the package_info_plus plugin
+version: 0.1.0
+homepage: https://plus.fluttercommunity.dev/
+repository: https://github.com/fluttercommunity/plus_plugins/tree/main/packages/
+
+environment:
+  sdk: ">=2.7.0 <3.0.0"
+  flutter: ">=1.17.0 <2.0.0"
```
Comment (jpnurmi) [inline packages/package_info_plus_linux/pubspec.yaml:9]: update both

**Pull request merged**